sympy/relation/binrel.py

```diff
@@ -0,0 +1,329 @@
+
+    @property
+    def reversed(self):
+        """
+        Return the relationship with sides reversed.
+        """
+        revfunc = self.function.reversed
+        if revfunc is None:
+            if self.function.is_symmetric:
+                return self.function(self.lhs, self.rhs)
+            return self
+        return self.function.reversed(self.rhs, self.lhs)
+
+    @property
+    def reversedsign(self):
+        """
+        Return the relationship with signs reversed.
+        """
+        revfunc = self.function.reversed
+        if revfunc is None:
+            return self
+        a, b = self.arguments
+        if not (isinstance(a, BooleanAtom) or isinstance(b, BooleanAtom)):
+            return self.function.reversed(-self.lhs, -self.rhs)
+        else:
+            return self
+
+    @property
+    def canonical(self):
+        """
+        Return a canonical form of the relational by putting a
+        number on the rhs, canonically removing a sign or else
+        ordering the args canonically. No other simplification is
+        attempted.
+        """
+        args = self.arguments
+        r = self
+        if r.rhs.is_number:
+            if r.rhs.is_Number and r.lhs.is_Number and r.lhs > r.rhs:
+                r = r.reversed
+        elif r.lhs.is_number:
+            r = r.reversed
+        elif tuple(ordered(args)) != args:
+            r = r.reversed
+
+        LHS_CEMS = getattr(r.lhs, 'could_extract_minus_sign', None)
+        RHS_CEMS = getattr(r.rhs, 'could_extract_minus_sign', None)
+
+        if isinstance(r.lhs, BooleanAtom) or isinstance(r.rhs, BooleanAtom):
+            return r
+
+        # Check if first value has negative sign
+        if LHS_CEMS and LHS_CEMS():
+            return r.reversedsign
+        elif not r.rhs.is_number and RHS_CEMS and RHS_CEMS():
+            # Right hand side has a minus, but not lhs.
+            # How does the expression with reversed signs behave?
+            # This is so that expressions of the type
+            # Q.eq(x, -y) and Q.eq(-x, y)
+            # have the same canonical representation
+            expr1, _ = ordered([r.lhs, -r.rhs])
+            if expr1 != r.lhs:
+                return r.reversed.reversedsign
+
+        return r
+
+    @property
+    def binary_symbols(self):
+        return set()
+
+    def _eval_ask(self, assumptions):
+        # quick exit for structurally same arguments
+        ret = self.function._compare_reflexive(self.lhs, self.rhs)
+        if ret is not None:
+            return ret
+
+        lhs, rhs = self.lhs.refine(assumptions), self.rhs.refine(assumptions)
+        r = self.function(lhs, rhs).simplify()
+
+        # attempt quick exit again with simplified arguments
+        ret = r.function._compare_reflexive(r.lhs, r.rhs)
+        if ret is not None:
+            return ret
+
+        # use multipledispatch handler
+        return r.function.eval(r.arguments, assumptions)
+
+    def _eval_simplify(self, **kwargs):
```

Comment:
This function seems to be working on the assumption that the relation is one of the standard ones (=, <, <=, >, >=).  The logic here wouldn't be correct for [this relation](https://github.com/sympy/sympy/pull/20723/files#diff-528c5c2d2eaeaeba2fcbec446816c839cd04afabc1bd241a60ecd1ac007a8952R38) for instance.

I'm also unclear what the difference is between this `_eval_simplify` and the one defined in `Equal` below.

Reply:
`Equal._eval_simplify` was design mistake. I enhanced simplification logic.